File: Semestre 4/IASA_Aux/IASA/2020_2021/tp1/src/src/reacao/Reaccao.java
package reacao;

public class Reaccao implements Comportamento{
	private Estimulo estimulo;
	private Accao resposta;
	
	public Reaccao(Estimulo estimulo, Accao resposta){
		this.estimulo = estimulo;
		this.resposta = resposta;
	}
	
	public Accao ativar(Estimulo estimuloEntrada) {
		//this.estimulo == estimulo ? acao:null 
		if(this.estimulo.equals(estimuloEntrada)) { //TODO: Override equals
			return resposta;
		}
		return null;
	}
}

File: Semestre 4/IASA_Aux/IASA/2022_2023/iasa45741/iasa-agente/src/lib/ecr/reaccao.py
from ecr.comportamento import Comportamento

class Reaccao(Comportamento):
    def __init__(self, estimulo, resposta):
        self._estimulo = estimulo
        self._resposta = resposta

    # O agente reage a percepcao recebe/sente do ambiente através do estimulo
    # esta reacção vai dar numa acao ativada pela resposta
    def activar(self,percepcao):
        intensidade = self._estimulo.detectar(percepcao) 
        if intensidade > 0:
                return self._resposta.activar(percepcao,intensidade) 


